sunscreen/src/types/zkp/gadgets/binary.rs

```diff
@@ -33,7 +33,7 @@ impl Gadget for ToUInt {
             return Err(ZkpError::gadget_error("Cannot create 0-bit uint."));
         }
 
-        if *val > BigInt::ONE.shl_vartime(self.n) {
+        if *val >= BigInt::ONE.shl_vartime(self.n) {
```

Comment:
Good catch! This is actually still incorrect when `n == 512`, however, as `BigInt::ONE.shl_vartime(512) == BigInt::ZERO`. I believe the correct fix is to change this line to

```suggestion
        if *val > BigInt::ONE.shl_vartime(self.n).wrapping_sub(&BigInt::ONE) {
```